pkg/controller/nodes/executor.go

```diff
@@ -353,17 +365,19 @@ func (c *nodeExecutor) handleQueuedOrRunningNode(ctx context.Context, nCtx *node
 		} else {
 			c.metrics.UnknownErrorDuration.Observe(ctx, startTime, endTime)
 		}
+		// When a node fails, we fail the workflow. Independent of number of nodes succeeding/failing, whenever a first node fails,
+		// the entire workflow is failed.
+		if np == v1alpha1.NodePhaseFailing {
```

Comment:
Is this assuming that a node-error will be a workflow error and hence we can count it so?

Reply:
Yes

Reply:
there is one problem maybe with this, if we allow run to completion flag, like an issue that was posted. But, I have to think more